src/snowflake/snowpark/_internal/data_source/drivers/pymsql_driver.py

```diff
@@ -0,0 +1,250 @@
+
+        processed_raw_schema = []
+        for type, col in zip(raw_types, raw_schema):
+            new_col = list(col)
+            new_col[1] = PymysqlTypeCode((new_col[1], type))
+            processed_raw_schema.append(new_col)
+
+        self.raw_schema = processed_raw_schema
+        return self.to_snow_type(processed_raw_schema)
+
+    def to_snow_type(self, schema: List[Any]) -> StructType:
+        """
+        pymysql mysql type to type code mapping:
+        https://github.com/PyMySQL/PyMySQL/blob/main/pymysql/constants/FIELD_TYPE.py
+
+        mysql type to snowflake type mapping:
+        https://other-docs.snowflake.com/en/connectors/mysql6/view-data#mysql-to-snowflake-data-type-mapping
+        """
+        fields = []
+        for col in schema:
+            (
+                name,
+                type_code,
+                display_size,
+                internal_size,
+                precision,
+                scale,
+                null_ok,
+            ) = col
+            snow_type = BASE_PYMYSQL_TYPE_TO_SNOW_TYPE.get(type_code, None)
+            if snow_type is None:
+                raise NotImplementedError(f"mysql type not supported: {type_code}")
+            if type_code in (PymysqlTypeCode.DECIMAL, PymysqlTypeCode.NEWDECIMAL):
+                # we did -2 here because what driver returned is precision + 2, mysql store + 2 precision internally
+                precision -= 2
+                if not self.validate_numeric_precision_scale(precision, scale):
+                    logger.debug(
+                        f"Snowpark does not support column"
+                        f" {name} of type {type_code} with precision {precision} and scale {scale}. "
+                        "The maximum number of digits in DECIMAL format for MySQL is 65. "
+                        "For Snowflake, the maximum is 38."
+                        "Supported up to the maximum allowed digits in Snowflake. When exceeded, precision is lost."
+                    )
+                    precision, scale = None, None
+                data_type = snow_type(
+                    precision if precision is not None else 38,
+                    scale if scale is not None else 0,
+                )
+            elif type_code == PymysqlTypeCode.TIMESTAMP:
+                data_type = snow_type(TimestampTimeZone.TZ)
+            elif type_code == PymysqlTypeCode.DATETIME:
+                data_type = snow_type(TimestampTimeZone.NTZ)
+            else:
+                data_type = snow_type()
+            fields.append(StructField(name, data_type, null_ok))
+        return StructType(fields)
+
+    def udtf_class_builder(self, fetch_size: int = 1000) -> type:
+        create_connection = self.create_connection
+
+        class UDTFIngestion:
+            def process(self, query: str):
+                def hexify_binary(row):
+                    return tuple(v.hex() if isinstance(v, bytes) else v for v in row)
```

Comment:
I think we can push the hex op to mysql: https://dev.mysql.com/doc/refman/8.4/en/string-functions.html#function_hex in mysql dialect

this function should work for binary col